src/main/java/io/numaproj/numaflow/function/FunctionService.java

```diff
@@ -296,14 +296,13 @@ private Udfunction.DatumResponseList buildDatumListResponse(MessageTList message
         return datumListBuilder.build();
     }
 
-    // log the exception and exit if there are any uncaught exceptions.
+    // wrap the exception and let it be handled in the central error handling logic.
     private void handleFailure(CompletableFuture<Void> failureFuture) {
         new Thread(() -> {
             try {
                 failureFuture.get();
             } catch (Exception e) {
-                e.printStackTrace();
-                System.exit(1);
+                throw new RuntimeException("error in reduce fn", e);
             }
         }).start();
```

Comment:
Why are we throwing the exception inside the thread? it has no effect

Reply:
As documented in L299. Here we wrap all the exceptions happens at the custom user code (application layer) as a `RuntimeException` and let it handled in the global grpc interceptor.

Reply:
since we are throwing the exception inside the thread, the global grpc interceptor won't get the exception.

Reply:
Hmm, but I wrote the test to ensure the grpc interceptor indeed get the exception.

Reply:
Sorry, now I re-read your previous comment, I got your point. In this case, it seems we should handle this specifically in the `ReduceShutdownActor.shutdown()`? Also, I don't see a good way to test this?